components/net/src/config.rs

```diff
@@ -31,3 +44,13 @@ pub trait RouteAddrs {
 pub trait Shards {
     fn shards(&self) -> &Vec<u32>;
 }
+
+pub trait ToAddrString {
```

Comment:
Yeah, this was added so I didn't need to make changes to the zmq library to accept ToSocketAddrs for constructing their connections.

In the long run we're going to end up owning the Rust ZMQ crate it looks like since none of my PR's have landed and it hasn't been updated in a while. This will be dropped if we do that ;)